src/main/java/generics/Fruit.java

```diff
@@ -0,0 +1,18 @@
+package generics;
+
+public class Fruit {
+    private final String typeOfFruit;
+    private final Float weight;
```

Comment:
```suggestion
    private final float weight;
```
зазвичай простий тип використовується (окрім випадку, коли значення може бути null)